src/agent/onefuzz-agent/src/tasks/report/libfuzzer_report.rs

```diff
@@ -46,35 +52,57 @@ pub struct ReportTask {
 }
 
 impl ReportTask {
-    pub fn new(config: impl Into<Arc<Config>>) -> Self {
-        let config = config.into();
-
-        let working_dir = config.common.task_id.to_string();
-        let poller = InputPoller::new(working_dir);
+    pub fn new(config: Config) -> Self {
+        let poller = InputPoller::new();
+        let config = Arc::new(config);
 
         Self { config, poller }
     }
 
-    pub async fn run(&mut self) -> Result<()> {
-        if self.config.check_fuzzer_help {
-            let target = LibFuzzer::new(
-                &self.config.target_exe,
-                &self.config.target_options,
-                &self.config.target_env,
-            );
-            target.check_help().await?;
+    pub async fn local_run(&self) -> Result<()> {
```

Comment:
Do we still needs all these local_run ?
The goal of my previous comment was to allows us to reuse the same code for both the local and the connected scenario. Can we just call the normal run directly now ?

Reply:
I missed the local_run for libfuzzer_fuzz, that's been addressed.

For the others, yes.  These use different queue mechanisms.  It would be reasonable move to a unified queue monitoring trait with a storage-queue and ionotify implementations, but given all the other changes this PR integrates, I'd like that to be a follow-on PR.